src/backend/catalog/aclchk.c

```diff
@@ -2670,28 +2670,50 @@ List *getActionName(AclMode mask)
 bool fallBackToNativeCheck(AclObjectKind objkind, Oid obj_oid, Oid roleid)
 {
   //for heap table, we fall back to native check.
-  if(objkind == ACL_KIND_CLASS)
+  if (objkind == ACL_KIND_CLASS)
   {
     char relstorage = get_rel_relstorage(obj_oid);
-    if(relstorage == 'h')
+    if (relstorage == 'h')
     {
       return true;
     }
   }
+  else if (objkind == ACL_KIND_NAMESPACE)
+  {
+	//native check build-in schemas.
+    if (obj_oid == PG_CATALOG_NAMESPACE || obj_oid == PG_INFORMATION_SCHEMA_NAMESPACE
```

Comment:
Is there need to add gp_toolkit schema? Seems need?

_`gp_toolkit` is an administrative schema that contains external tables, views, and functions that you can access with SQL commands. All database users can access gp_toolkit to view and query the system log files and other system metrics._

refer: http://hdb.docs.pivotal.io/201/hawq/ddl/ddl-schema.html